jaxlib/rocm/hipsolver_kernels.cc

```diff
@@ -126,30 +126,30 @@ static absl::Status Potrf_(hipStream_t stream, void** buffers,
       case HipsolverType::F32: {
         JAX_RETURN_IF_ERROR(JAX_AS_STATUS(hipsolverSpotrfBatched(
             handle.get(), d.uplo, d.n, static_cast<float**>(workspace), d.n,
-            static_cast<float*>(workspace + (d.batch * sizeof(float*))), d.lwork,
-            info, d.batch)));
+            static_cast<float*>(static_cast<float*>(workspace) + (d.batch *
```

Comment:
The following changes fix the compilation error given for arithmetic with void pointers.

Reply:
I think this might be incorrect. If you've cast the base pointer to `float *`, then the pointer arithmetic shouldn't have a `sizeof(float)` scaling factor. So either cast the base pointer to `char*` or remove the `sizeof(float)` I think?

Reply:
So I tried casting the base pointer to char* and leaving the sizeof but the compiler gave an error:

error: static_cast from 'char *' to 'float *' is not allowed

Then I tried leaving the base pointer cast to float* and removing sizeof(float*) like so:

static_cast<float*>(static_cast<float*>(workspace) + d.batch

But in that case the linalg_test failed.

The way it is written seems to make everybody happy.

Reply:
I think you should have cast the base type to `float**`. i.e. the structure of the workspace is:
a) `batch` float* pointers followed by
b) a workspace of floats for the use of the kernel.

So your base pointer should be `float**`, and you should step `batch` elements into it.

Reply:
Also the fix for the `static_cast` error is you need to use `reinterpret_cast`.

Reply:
Fixed.  Thanks for your suggestions.